gpm/tests/test_utils/test_manipulations.py

```diff
@@ -181,21 +198,267 @@ def test_slice_range_with_valid_data(
+            a_3d_dataarray,
+            vmin=vmin,
+            vmax=vmax,
+        )
+
+    # Test with valid values
+    a_3d_dataarray.data[:, :, 1] = 11  # keep this layer
+    a_3d_dataarray.data[2, 2, 2] = 12  # layer kept even if single value is valid
+    a_3d_dataarray.data[2, 2, 3] = 21  # not valid
+    returned_da = manipulations.slice_range_where_values(
+        a_3d_dataarray,
+        vmin=vmin,
+        vmax=vmax,
+    )
+    expected_data = a_3d_dataarray.data[:, :, 1:3]
+    np.testing.assert_allclose(returned_da.data, expected_data)
+
+
+def test_slice_range_at_value(
+    a_3d_dataarray: xr.DataArray,
+) -> None:
+    """Test slice_range_at_value function"""
+
+    value = 100
+    returned_slice = manipulations.slice_range_at_value(a_3d_dataarray, value)
+    vertical_indices = np.abs(a_3d_dataarray - value).argmin(dim="height")
+    expected_slice = a_3d_dataarray.isel(height=vertical_indices).data
+    np.testing.assert_allclose(returned_slice.data, expected_slice)
+
+
+def test_slice_range_at_max_value(
+    a_3d_dataarray: xr.DataArray,
+) -> None:
+    """Test slice_range_at_max_value function"""
+
+    returned_slice = manipulations.slice_range_at_max_value(a_3d_dataarray)
+    expected_slice = a_3d_dataarray.isel(height=-1).data
+    np.testing.assert_allclose(returned_slice.data, expected_slice)
+
+
+def test_slice_range_at_min_value(
+    a_3d_dataarray: xr.DataArray,
+) -> None:
+    """Test slice_range_at_min_value function"""
+
+    returned_slice = manipulations.slice_range_at_min_value(a_3d_dataarray)
+    expected_slice = a_3d_dataarray.isel(height=0).data
+    np.testing.assert_allclose(returned_slice.data, expected_slice)
+
+
+@pytest.mark.parametrize("variable", ["airTemperature", "height"])
+def test_slice_range_at(
+    variable: str,
+    a_3d_dataarray: xr.DataArray,
+) -> None:
+    """Test slice_range_at_temperature and slice_range_at_height functions"""
+
+    manipulations_functions = {
+        "airTemperature": manipulations.slice_range_at_temperature,
+        "height": manipulations.slice_range_at_height,
+    }
+    manipulations_function = manipulations_functions[variable]
+
+    ds = xr.Dataset({variable: a_3d_dataarray})
+    value = 105
+    returned_slice = manipulations_function(ds, value)
+    expected_slice = a_3d_dataarray.isel(height=3).data
+    np.testing.assert_allclose(returned_slice[variable].data, expected_slice)
+
+
+def test_get_height_at_temperature(
+    a_3d_dataarray: xr.DataArray,
+) -> None:
+    """Test get_height_at_temperature function"""
+
+    da_temperature = a_3d_dataarray.copy()
+    da_height = a_3d_dataarray.copy()
+    da_height.data[:] = da_height.data[:] + 500
+
+    temperature = 105
+    returned_da = manipulations.get_height_at_temperature(da_height, da_temperature, temperature)
+    expected_data = da_height.data[:, :, 3]
+    np.testing.assert_allclose(returned_da.data, expected_data)
+
+
+def test_get_xr_dims_dict(
```

Comment:
I think the function `get_xr_dims_dict` can now be simply replaced by `ds.sizes` or `da.sizes` where it occurs across the codebase ;)